src/io_posix.c

```diff
@@ -0,0 +1,44 @@
+#include "io.h"
+
+ssize_t
+udx__sendmsg(udx_t *self, udx_packet_t *pkt) {
+  ssize_t size;
+  struct msghdr h;
+
+  memset(&h, 0, sizeof(h));
+
+  h.msg_name = &(pkt->dest);
+  h.msg_namelen = sizeof(pkt->dest);
+
+  h.msg_iov = (struct iovec *) &(pkt->buf);
+  h.msg_iovlen = pkt->buf[1].len ? 2 : 1;
```

Comment:
i like all your changes! i think we could consider having an int on the packet to indicate how iovlen in the future (saves us having to explicitly set this to 0 also, but more importantly more clear?)